### PR TITLE
fix: Safe version feature helper

### LINKED ISSUE #330: `enabledFeatures` returns all features for v1.0.0 Safes
## Description

`enabledFeatures` does not filter correctly.

## Steps to reproduce

Call `enabledFeatures('1.0.0')` and observe all `FEATURES` listed in array.

## Expected result

Filters correctly, e.g. for `'1.0.0'` an empty array is returned.

## Additional context

This can be fixed by using a string enum instead:

```ts
export enum FEATURES {
  ETH_SIGN = 'ETH_SIGN',
  SAFE_FALLBACK_HANDLER = 'SAFE_FALLBACK_HANDLER',
  SAFE_TX_GAS_OPTIONAL = 'SAFE_TX_GAS_OPTIONAL',
  SAFE_TX_GUARDS = 'SAFE_TX_GUARDS',
}
```

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves
Resolves #330

## How this PR fixes it

- `hasFeature` has been refactored/renamed to `hasSafeFeature` for future import on the frontend.
- `FEATURES` has also been renamed to `SAFE_FEATURES` and changed to a string enum in order to fix the filtering specified in the issue. The issue was originally fixed [here](https://github.com/safe-global/web-core/pull/1595).

### CODE REVIEW

**Comment by iamacook** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/safe-core-sdk/src/utils/safeVersions.ts [hunk 1]
```diff
@@ -1,31 +1,24 @@
...
+  [SAFE_FEATURES.SAFE_TX_GUARDS]: '>=1.3.0',
+  [SAFE_FEATURES.SAFE_FALLBACK_HANDLER]: '>=1.1.1',
+  [SAFE_FEATURES.ETH_SIGN]: '>=1.1.0'
 }
 
-const isEnabledByVersion = (feature: FEATURES, version: string): boolean => {
-  if (!(feature in FEATURES_BY_VERSION)) {
-    return true
+// Note: gatewau returns `SafeInfo['version']` as `null` for unsupported contracts
+export const hasSafeFeature = (feature: SAFE_FEATURES, version: string | null): boolean => {
```
Comment (germartinez) [inline packages/safe-core-sdk/src/utils/safeVersions.ts:18]: I would make `version` a required parameter. If it is `undefined` or `null` it should be handled outside of the function.
```suggestion
export const hasSafeFeature = (feature: SAFE_FEATURES, version: string): boolean => {
```

File: packages/safe-core-sdk/src/utils/safeVersions.ts [hunk 2]
```diff
@@ -1,31 +1,24 @@
...
+  if (!version || !(feature in SAFE_FEATURES_BY_VERSION)) {
```
Comment (germartinez) [inline packages/safe-core-sdk/src/utils/safeVersions.ts:19]:
```suggestion
  if (!(feature in SAFE_FEATURES_BY_VERSION)) {
```

**Review by germartinez — CHANGES_REQUESTED** [review]

**Comment by germartinez** [comment]
It was a good catch!
Thank you @iamacook!
:rocket:

**Pull request merged**